### DISCUSSION TITLE
Samsung Refrigerator?

### DISCUSSION THREAD

**Discussion by Milkysunshine** [theengs/decoder#438] [New devices/things integration]
In trying to get info on the Govee H5179, I see my Samsung Refrigerator is also spitting out some gibberish at me. (gibberish to me at least)

```
DEBUG:BLEGateway:50:FD:D5:4E:76:02:AdvertisementData(local_name='Refrigerato', manufacturer_data={117: b'942V\x04\x02\x04\x00'}, rssi=-54)
DEBUG:BLEGateway:Sent `{"manufacturerdata": "75003934325604020400", "name": "Refrigerato", "id": "50:FD:D5:4E:76:02", "rssi": -54, "mfr": "Samsung Electronics Co. Ltd."}` to topic `home/TheengsGateway/BTtoMQTT/50FDD54E7602`
```

I have no idea if there is current temperatures or what in there, but I can gladly provide the limited information the refrigerator actually provides to figure it out.

Just let me know what is needed and I'll gladly help.

**Comment by DigiH** [comment]
If you don't actually have a thermometer which could go inside the fridge to get a rough idea for the temperature, you could also put in some fresh warm bottle of soft drink, milk or whatever which would get the temperature up for a while before it cools down again, then some additional readings would be needed, to see if and how

`"manufacturerdata": "75003934325604020400"`

changes at all. Also if there are different settings available on the fridge, changing these and seeing if it changes the manufacturerdata will give pointers if and which data might be encoded there.

**Comment by Milkysunshine** [comment]
I actually have all 3 of my H5179s in the fridge currently. I had issues with it when it was brand new and needed to keep track to make sure the food isn't getting too warm. They came and replaced parts and have seemingly resolved the issue, but I leave them in there anyway for now.  The 3 shelves are vastly different in temperature, but I'll see what I can find.

I will do as you suggested and mess with the ice maker, temp settings, pitcher auto fill, and document if and what changes.

**Comment by 1technophile** [comment]
> "name": "Refrigerato"

For our understanding did you set this name or is it the default one ?

**Reply by Milkysunshine** [reply]
> > "name": "Refrigerato"
> 
> For our understanding did you set this name or is it the default one ?

I did set that name in the smart things app, but changing it in the app did not immediately change the local_name displayed in the BLEGateway scans.

**Comment by Milkysunshine** [comment]
The 75003934325604020400 doesn't change by changing any settings.

I do occasionally see a MUCH longer string sent. It is quite infrequent. Changing settings on the fridge doesn't seem to have any immediate effect on either string. I will check them for changes.


```
DEBUG:BLEGateway:50:FD:D5:4E:76:02:AdvertisementData(local_name='Refrigerato', manufacturer_data={117: b'B\x0c\x83\x03]0AJTRE0\x02\x01\x01P\xfd\xd5Nv\x02\n\x02\x04'}, rssi=-59)
DEBUG:BLEGateway:Sent `{"manufacturerdata": "7500420c83035d30414a5452453002010150fdd54e76020a0204", "name": "Refrigerato", "id": "50:FD:D5:4E:76:02", "rssi": -59, "mfr": "Samsung Electronics Co. Ltd."}` to topic `home/TheengsGateway/BTtoMQTT/50FDD54E7602`
```

**Reply by DigiH** [reply]
Good find with the longer manufacturerdata, it also includes the device's MAC address, but possibly some some more encoded data.

I suppose the main things of interested would be the temperature and humidity.

As the temperature does change periodically when colling down while the compressor is working, then slowly warming up again, the thing to watch out for are any small changes in the short or long manufacturerdata.

If they always stay exactly the same it could also well be that the fridge does not braodcasts its data vie the BLE advertising data, and that it would require a connection, which is out of the scope of Decoder.

Let see how things look with more longer manufacturerdata samples.

**Reply by Milkysunshine** [reply]
"7500420c83035d30414a5452453002010150fdd54e76020a0204" does not seem to change either. I looked at about 40 entries over the 
past 2+ hours and neither change. 

Oh well, it was worth a shot.

Maybe it is for a tech to do diagnostics? Since it isn't changing, it doesn't appear to be any kind of status regarding settings or values that would be of use to an average human. I know the fridge has wireless, and that is how the smart things app interacts with it. I'm curious what the Bluetooth is even for...

**Reply by DigiH** [reply]
This are likely then just a Bluetooth becons for pairing recognition, and then for the Samsung App to retrieve data through connection. 

This is not currently in the scope of Theengs Decoer, which can only receive and decode freely broadcast advertising data.

And as you said, well worth testing it out. So if you ever see any advertising data from other devices you might have in the future, which do change, it is worth letting us know so we can investigate further together.
